### DISCUSSION TITLE
RequiredSkill checks are too generous.

### DISCUSSION THREAD

**Discussion by DarkStatic** [FakeFishGames/Barotrauma#11113] [Ideas and suggestions]
Now that `OnSuccess` has a distinction from `OnUse` I've been able to test the success rate of certain medical items: Even with about half the required skill level to use a certain item, the item rarely ever "fails" to be used. In about 20 consecutive uses of said item with a medical skill of 15, and a required skill of 30, failiure only occurred 3 times.

To better highlight the separation of `OnSuccess` and `OnFailiure` I propose the measure of probability is adjusted so that the success rate isn't so generous for unskilled users. Additionally, I'd like for there to be an option for `RequiredSkill` to be non-probability base, so that they may act as hard limiters for more dedicated items.

**Comment by BiffW** [comment]
Just a reminder. If you want hard limiter. You can just copy the skill check conditional defined in "hmg". It stuns the shooter 100% even if the shooter has 74 weapons skill which is just 1 point less than required skill.

**Comment by ubertpendragon** [comment]
I will also add that crafting checks for "good" quality are also very generous.

Currently the formula for crafting Good quality is: 

`Required level  + (100-Required Level)*0.2`

Which means you get good items very early, sometimes even with skill levels given at spawn.

Mechanics and Engineers get a base mechanical/electrical skill of 40 to 50, they are almost instantly capable of crafting Good items

-20 skill item gets good at 36 (oxygen/welding fuel tanks, wrench/screwdriver)
-30 skill item at 44 (welding tools)
-40 skill item at 52

Turn that 0.2 into a 0.4?
The progression curve isn't smooth, considering you have to wait 100 skill for Excellent.
